### PR TITLE
c: use WARN_UNUSED for ByteExtract* functions

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues/3658) ticket:
 Previous PR: #8014
Describe changes:

- Add WARN_UNUSED for all ByteExtract functions in util-byte.h
- Fix warnings generated in code due to the macro

### CODE REVIEW

File: src/util-byte.h [hunk 1]
```diff
@@ -226,7 +226,7 @@ int ByteExtractStringInt64(int64_t *res, int base, size_t len, const char *str);
  * \return n Number of bytes extracted on success
  * \return -1 On error
  */
-int ByteExtractStringInt32(int32_t *res, int base, size_t len, const char *str);
+int  ByteExtractStringInt32(int32_t *res, int base, size_t len, const char *str);
```
Comment (jufajardini) [inline src/util-byte.h:229]: I think a `WARN_UNUSED` was missed here and was present in the previous PR.

File: src/util-byte.h [hunk 2]
```diff
@@ -464,7 +464,7 @@ int WARN_UNUSED StringParseI16RangeCheck(
  * \param len Number of bytes to extract (23 max or 0 for unbounded)
  * \param str String to extract from
  *
- * \return n Number of bytes extracted on success
+ * \return n Number of bytes extracted on su./configccess
```
Comment (jufajardini) [inline src/util-byte.h:467]: I think this was forgotten, too! :P

**Review by jufajardini — CHANGES_REQUESTED** [review]
I think one change was missed and is not showing from the previous PR to this one.

And one accidental change also still needs fix, I think.

(see inline comments :) )

File: src/util-byte.h (hunk 1 above)
Comment (hsadia538) [reply src/util-byte.h:229]: 🫣 Thanks for catching this.

**Pull request closed without merging**